.github/actions/setup-go/actions.yaml

```diff
@@ -0,0 +1,27 @@
+name: 'Setup golang with master only caching'
```

Comment:
Probably stupid question: what is the benefit of doing this?

Reply:
If you look at our Actions caches https://github.com/k3s-io/k3s/actions/caches, we keep going over our 10GB limit with a bunch of `setup-go-Linux-ubuntu22-go-1.21.7-bda3a8cce4ad2d34dfb706b901c4ce11eab6d77c2d08dd8f0eae8694a7a27733 ` caches. These are all from various PRs that get opened (hence the SHA section at the end of the name). These caches are ONLY valid for the PR that opened them for security reasons (cache poisoning prevention).

Reply:
The idea is to only allow `master` branch to Save a valid cache entry, because that cache is usable by all PRs, as along as the checksum is the same (ie no changes to go.mod).

Reply:
Thanks Derek